src/main/java/de/srendi/advancedperipherals/common/util/SwarmEventDispatcher.java

```diff
@@ -0,0 +1,76 @@
+package de.srendi.advancedperipherals.common.util;
+
+import dan200.computercraft.api.peripheral.IComputerAccess;
+import dan200.computercraft.shared.computer.core.ServerContext;
+import dan200.computercraft.shared.computer.core.ServerComputer;
+import de.srendi.advancedperipherals.AdvancedPeripherals;
+import de.srendi.advancedperipherals.lib.peripherals.BasePeripheral;
+import net.minecraft.server.MinecraftServer;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.jetbrains.annotations.NotNull;
+
+@Mod.EventBusSubscriber(modid = AdvancedPeripherals.MOD_ID)
+public final class SwarmEventDispatcher {
```

Comment:
Could you maybe add some java docs to this to explain the goals and how that system works?